### ISSUE TITLE
Python 3.8 issue for code objects due to position-only arguments

### ISSUE THREAD

**Issue by irmen** [uqfoundation/dill#318]
Dill has at least the following compatibility problem with Python 3.8  (which just released the first beta version):

```
Python 3.8.0b1+ (heads/3.8:d74438b633, Jun  5 2019, 22:17:23) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dill
>>> def func(x): pass
... 
>>> p=dill.dumps(func)   
>>> dill.loads(p)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/irmen/venv/py38/lib/python3.8/site-packages/dill/_dill.py", line 317, in loads
    return load(file, ignore)
  File "/home/irmen/venv/py38/lib/python3.8/site-packages/dill/_dill.py", line 305, in load
    obj = pik.load()
TypeError: an integer is required (got type bytes)
>>> 

```

**Comment by mmckerns** [comment]
Yuck.  Thanks for the ping.  I was waiting for the beta to drop, and apparently it has.

**Comment by irmen** [comment]
Indeed it has! I was using it to validate my own projects against it.  One test failed due to the dill dependency.

**Comment by mmckerns** [comment]
I see, the issue above is due to changes to accommodate position-only functions.

**Comment by mmckerns** [comment]
The above PR fixes this issue.  Thanks for the poke. Please open another PR if you find something else.

**Comment by mmckerns** [comment]
@irmen: new release was cut earlier this week.  I'll continue to work on any issues that arise with changes from 3.8.  All tests pass, but I'm sure there are issues lurking.

**Comment by irmen** [comment]
@mmckerns can confirm green status on Pyro4's python 3.8 build :) thanks!

**Comment by metya** [comment]
I have this error on my `python 3.8.1` and `dill 0.3.1.1`

**Comment by Huyston** [comment]
I still have this issue with python 3.8.1 and dill 0.3.1.1

```
Traceback (most recent call last):
  File "main.py", line 988, in openFile
    self.archive[docID] = pickle.load(archive)
  File "/usr/lib/python3.8/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/usr/lib/python3.8/site-packages/dill/_dill.py", line 472, in load
    obj = StockUnpickler.load(self)
TypeError: an integer is required (got type bytes)
```

**Comment by irmen** [comment]
maybe ping @mmckerns about this because this issue has been closed already a while ago

**Comment by PranY** [comment]
I'm facing the exact same issue.
Python 3.8.2
dill 0.3.1.1

```
TypeError                       Traceback (most recent call last)
<ipython-input-14-eb273f9c9239> in <module>
----> 1 test = torch.load('test.pth', pickle_module=dill)

~/.local/lib/python3.8/site-packages/torch/serialization.py in load(f, map_location, pickle_module, **pickle_load_args)
    591                     return torch.jit.load(f)
    592                 return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
--> 593         return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
    594 
    595 

~/.local/lib/python3.8/site-packages/torch/serialization.py in _legacy_load(f, map_location, pickle_module, **pickle_load_args)
    771     unpickler = pickle_module.Unpickler(f, **pickle_load_args)
    772     unpickler.persistent_load = persistent_load
--> 773     result = unpickler.load()
    774 
    775     deserialized_storage_keys = pickle_module.load(f, **pickle_load_args)

~/.local/lib/python3.8/site-packages/dill/_dill.py in load(self)
    470 
    471     def load(self): #NOTE: if settings change, need to update attributes
--> 472         obj = StockUnpickler.load(self)
    473         if type(obj).__module__ == getattr(_main_module, '__name__', '__main__'):
    474             if not self._ignore:

TypeError: an integer is required (got type bytes)
```

**Comment by garrettbingham** [comment]
Upgrading dill to 0.3.3 solved the issue for me with python 3.8.0

**Comment by mmckerns** [comment]
@garrettbingham: I'm seeing the same results as you are.  @Huyston: can you confirm?  If so, I'll keep the issue closed.
